File: Emulator/src/main/content/global/skill/crafting/items/armour/capes/Dyes.java
package content.global.skill.crafting.items.armour.capes;

import core.game.node.item.Item;
import org.rs.consts.Items;

import java.util.HashMap;
import java.util.Map;

/**
 * The enum Dyes.
 */
public enum Dyes {
    /**
     * The Black.
     */
    BLACK(new Item(Items.BLACK_MUSHROOM_INK_4622)),
    /**
     * The Red.
     */
    RED(new Item(Items.RED_DYE_1763)),
    /**
     * The Yellow.
     */
    YELLOW(new Item(Items.YELLOW_DYE_1765)),
    /**
     * The Blue.
     */
    BLUE(new Item(Items.BLUE_DYE_1767)),
    /**
     * The Orange.
     */
    ORANGE(new Item(Items.ORANGE_DYE_1769)),
    /**
     * The Green.
     */
    GREEN(new Item(Items.GREEN_DYE_1771)),
    /**
     * The Purple.
     */
    PURPLE(new Item(Items.PURPLE_DYE_1773)),
    /**
     * The Pink.
     */
    PINK(new Item(Items.PINK_DYE_6955));

    private final Item item;

    private static final Map<Integer, Dyes> itemToDyeMap = new HashMap<>();

    static {
        for (Dyes dye : values()) {
            itemToDyeMap.put(dye.item.getId(), dye);
        }
    }

    Dyes(Item item) {
        this.item = item;
    }

    /**
     * Gets item.
     *
     * @return the item
     */
    public Item getItem() {
        return item;
    }

    /**
     * For item dyes.
     *
     * @param item the item
     * @return the dyes
     */
    public static Dyes forItem(Item item) {
        return itemToDyeMap.get(item.getId());
    }
}

File: Emulator/src/main/content/global/skill/crafting/items/armour/capes/Dyes.kt
package content.global.skill.crafting.items.armour.capes

import core.game.node.item.Item
import org.rs.consts.Items

/**
 * The dyes.
 */
enum class Dyes(val item: Item) {
    BLACK(Item(Items.BLACK_MUSHROOM_INK_4622)),
    RED(Item(Items.RED_DYE_1763)),
    YELLOW(Item(Items.YELLOW_DYE_1765)),
    BLUE(Item(Items.BLUE_DYE_1767)),
    ORANGE(Item(Items.ORANGE_DYE_1769)),
    GREEN(Item(Items.GREEN_DYE_1771)),
    PURPLE(Item(Items.PURPLE_DYE_1773)),
    PINK(Item(Items.PINK_DYE_6955));

    companion object {
        private val itemToDyeMap: MutableMap<Int?, Dyes?> = HashMap()

        init {
            for (dye in values()) {
                itemToDyeMap[dye.item.id] = dye
            }
        }

        /**
         * For item dyes.
         *
         * @param item the item
         * @return the dyes
         */
        @JvmStatic
        fun forItem(item: Item): Dyes? {
            return itemToDyeMap[item.id]
        }
    }
}
